core/src/autogluon/core/pseudolabeling/pseudolabeling.py

```diff
@@ -0,0 +1,218 @@
+        y_pred_proba_max = y_pred_proba_og.max(axis=1)
+        curr_threshold = threshold
+        curr_percentage = (y_pred_proba_max >= curr_threshold).mean()
+        num_rows = len(y_pred_proba_max)
+
+        if curr_percentage > max_proportion_prob or curr_percentage < min_proportion_prob:
+            if curr_percentage > max_proportion_prob:
+                num_rows_threshold = max(np.ceil(max_proportion_prob * num_rows), 1)
+            else:
+                num_rows_threshold = max(np.ceil(min_proportion_prob * num_rows), 1)
+            curr_threshold = y_pred_proba_max.sort_values(ascending=False).iloc[int(num_rows_threshold) - 1]
+
+        test_pseudo_indices = (y_pred_proba_max >= curr_threshold)
+        test_pseudo_indices = sample_bins_uniformly(y_pred_proba=y_pred_proba_og, df_indexes=test_pseudo_indices)
+    else:
+        test_pseudo_indices = pd.Series(data=False, index=y_pred_proba_og.index)
+        test_pseudo_indices_true = test_pseudo_indices.sample(frac=proportion_sample, random_state=0)
+        test_pseudo_indices[test_pseudo_indices_true.index] = True
+
+    test_pseudo_indices = test_pseudo_indices[test_pseudo_indices]
+
+    return test_pseudo_indices
+
+
+def filter_ensemble_pseudo(predictor, unlabeled_data: pd.DataFrame, num_models: int = 5):
+    """
+    Uses ensemble filtering for pseudo labeling
+
+    Parameters:
+    -----------
+    predictor: Fitted tabular predictor
+    unlabeled_data: Unlabeled data for top k models to predict on
+    num_models: Number of top models to ensemble
+
+    Returns:
+    -------
+    pd.Series of indices that met pseudo labeling requirements
+    """
+    original_k = num_models
+    leaderboard = predictor._trainer.leaderboard()
+    num_models = len(leaderboard)
+
+    if num_models < 2:
+        raise Exception('Ensemble pseudo labeling was enabled, but only one model was trained')
+
+    if num_models != num_models:
+        logging.warning(f'Ensemble pseudo labeling expected {original_k}, but only {num_models} fit.')
+
+    if predictor.problem_type in PROBLEM_TYPES_CLASSIFICATION:
+        return filter_ensemble_classification(predictor=predictor, unlabeled_data=unlabeled_data,
+                                              leaderboard=leaderboard, num_models=num_models)
+    else:
+        return filter_pseudo_std_regression(predictor=predictor, unlabeled_data=unlabeled_data,
+                                            leaderboard=leaderboard, num_models=num_models)
+
+
+def filter_pseudo_std_regression(predictor, unlabeled_data: pd.DataFrame, num_models, leaderboard,
+                                 lower_bound: float = -0.25, upper_bound: float = 0.25):
+    """
+    Predicts on unlabeled_data using the top k models. Then gets standard deviation of each
+    row's prediction and the standard deviation across all rows of standard deviations and
+    incorporates rows with a z-score in the range of lower_bound and upper_bound.
```

Comment:
description is unclear. First off should clarify that this is: standard deviation of each row's prediction across the top k models. Second should clarify how z-score is computed.

Reply:
added more detail